src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
-      graphics.setTransform(original);
+    // set thickness and color of the future drawings
+    graphics.setStroke(new BasicStroke(3.5f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
+    graphics.setPaint(Color.red);
+    graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    if (routeDescription == null) {
+      return;
+    }
+    final Route route = routeDescription.getRoute();
+    if (route == null) {
+      return;
+    }
+    final int numTerritories = route.getAllTerritories().size();
+    final Point[] points = getRoutePoints(routeDescription, mapData);
+    final boolean tooFewTerritories = numTerritories <= 1;
+    final boolean tooFewPoints = points.length <= 2;
+    final int xOffset = view.getXOffset();
+    final int yOffset = view.getYOffset();
+    final double scale = view.getScale();
+    final int jointsize = 10;
+    if (tooFewTerritories || tooFewPoints) {
+      drawLineWithTranslate(graphics, new Line2D.Float(routeDescription.getStart(), routeDescription.getEnd()), xOffset,
+          yOffset, scale);
+      graphics.fillOval((int) (((routeDescription.getEnd().x - xOffset) - jointsize / 2) * scale),
+          (int) (((routeDescription.getEnd().y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
+    } else {
+      drawCurvedPath(graphics, points, view);
+      drawMoveLength(graphics, routeDescription, points, xOffset, yOffset, view, mapData.scrollWrapX(),
+          mapData.scrollWrapY(), numTerritories, movementLeftForCurrentUnits);
+    }
+    drawJoints(graphics, points, xOffset, yOffset, jointsize, scale);
+    final Image cursorImage = routeDescription.getCursorImage();
+    if (cursorImage != null) {
+      graphics.drawImage(cursorImage,
+          (int) (((routeDescription.getEnd().x - xOffset) - (cursorImage.getWidth(null) / 2)) * scale),
+          (int) (((routeDescription.getEnd().y - yOffset) - (cursorImage.getHeight(null) / 2)) * scale), null);
     }
   }
 
-  private static void drawWithTranslate(final Graphics2D graphics, final Shape shape, final double translateX,
-      final double translateY) {
-    if (shape instanceof Ellipse2D.Double) {
-      Ellipse2D.Double elipse = (Ellipse2D.Double) shape;
-      elipse = new Ellipse2D.Double(elipse.x + translateX, elipse.y + translateY, elipse.width, elipse.height);
-      graphics.draw(elipse);
+  private static void drawJoints(Graphics2D graphics, Point[] points, int xOffset, int yOffset, int jointsize,
+      double scale) {
+    for (Point p : points) {
+      graphics.fillOval((int) (((p.x - xOffset) - jointsize / 2) * scale),
+          (int) (((p.y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
     }
-    if (shape instanceof Polygon) {
-      ((Polygon) shape).translate((int) translateX, (int) translateY);
-      graphics.fill(shape);
-      ((Polygon) shape).translate((int) -translateX, (int) -translateY);
+
+  }
+
+  protected static double[] getIndex(Point[] points) {
```

Comment:
The name of this method does not match the functionality. The name of the helper method can be a bit more detailed and longer to help describe what is happening.